### ISSUE TITLE
test set은 왜 딱 한번만 해야하지 궁금합니다.

### ISSUE THREAD

**Issue by kimkc** [liganega/CodingRG#29]
test set을 가지고 Gradient Descent Optimizer를 W가 업데이트되는 것도, cost 값을 업데이트하는 것도 아니고, accuracy에서 Hypothesis로 돌려서 나오는 Y' 값을 가지고 실제 Y값과 비교하는 것인데 이것이 학습에 어떤 영향을 주길래 한번밖에 안 되는지 궁금합니다.

**Comment by SWEETCANDY1008** [comment]
training set을 가지고 학습을 시킨 후 나온 학습 결과가 잘 학습이 되었는지를 확인하기 위해 test set을 사용합니다.

다시 말하면 training set으로 학습이 끝나면, test set을 이용해서, 학습된 알고리즘이 얼마나 정확히 예측(Prediction)하는지를 측정할 수 있습니다.

예) 수능을 보기 위해 수 없이 많은 공부를 통해 학습하였고(training set), 중간 중간 모의고사를 봐왔으며(validation set), 마지막으로 그동안의 학습을 확인하기 위해 수능을 봅니다(test set)

그래서 우리가 예측한 모델이 얼마나 정확히 예측하는지를 확인하기 위해 test set을 딱 한번만 돌리게 되고, 예측이 정확하지 않을 경우 또 다시 training set을 가지고 학습을 시키고 다시 test set을 통해 예측이 정확한지 확인한다고 생각합니다.

**Comment by liganega** [comment]
아래 링크 비디오에 답변 있음

https://youtu.be/aOyWv2Ynzuc
